### PR TITLE
Set the job name before a suite is run

### PR DESCRIPTION
**Pull request by gabriel-kohen-by**
## Proposed changes

Per https://github.com/webdriverio/webdriverio/issues/7609 we need to account for suite `before` logic will be long running or failing and we want to give intication early on running the suite's tests of what is running.

## Types of changes

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

## Checklist

- [X] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/main/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have added proper type definitions for new commands (if appropriate)

### CODE REVIEW

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Thanks for taking a stab at this!

Let's add one or multiple test cases [here](https://github.com/gabriel-kohen-by/webdriverio/blob/main/packages/wdio-sauce-service/tests/service.test.ts#L60-L66) and ensure that the if statement is only executed if situation is as expected.

**Comment by gabriel-kohen-by** [comment]
@christian-bromann, thanks for the feedback. I've tweaked and added some assertions to account for the new logic. Let me know if there is anything I need to change.

File: packages/wdio-sauce-service/tests/service.test.ts [hunk 1]
```diff
@@ -485,7 +507,7 @@ test('_uploadLogs should not fail in case of a platform error', async () => {
     const service = new SauceService(
         {},
         {},
-        { outputDir: '/foo/bar' } as any
+            { outputDir: '/foo/bar' } as any
```
Comment (christian-bromann) [inline packages/wdio-sauce-service/tests/service.test.ts:510]:
```suggestion
        { outputDir: '/foo/bar' } as any
```

**Review by christian-bromann — APPROVED** [review]
LGTM 👍  thanks a lot!

**Comment by gabriel-kohen-by** [comment]
@christian-bromann , not sure why the e2e failed on the Mac. Can't think the changes are related to this failure. Is the e2e flaky in nature?

**Comment by christian-bromann** [comment]
@gabriel-kohen-by yeah, the windows e2e tests aren't that stable atm. Will go ahead and merge. Thanks a lot!

**Comment by gabriel-kohen-by** [comment]
Thanks for merging @christian-bromann. How does the release process works as far as making it to GA?

**Comment by christian-bromann** [comment]
I just need to kick off a GitHub action to release a new version. Will do so now.

**Comment by wswebcreation** [comment]
This breaks setting the job name in Jasmine, because the `suiteTitle` is `Jasmine__TopLevel__Suite` and the `fullName` is `null`, so no alternative. Will fix this in another PR

**Pull request merged**